pkg/apis/kubermatic/v1/preset.go

```diff
@@ -114,6 +114,8 @@ func (s *PresetSpec) SetEnabled(enabled bool) {
 type ProviderPreset struct {
 	// Only enabled presets will be available in the KKP dashboard.
 	Enabled *bool `json:"enabled,omitempty"`
+	// Only editable presets will allow customization of non-essential fields in the KKP dashboard.
+	IsEditable *bool `json:"isEditable,omitempty"`
```

Comment:
```suggestion
	// IsCustomizable marks a preset as editable on the KKP UI; Customizable presets still have the credentials obscured on the UI, but other fields that are not considered private are displayed during cluster creation. Users can then update those fields, if required.
	// NOTE: This is only supported for OpenStack Cloud Provider in KKP 2.26. Support for other providers will be added later on. 
	IsCustomizable bool `json:"isCutomizable,omitempty"`
```

I'm still not a big fan of the variable name but I don't have better options either.